tests/bdd/bdd_test.go

```diff
@@ -0,0 +1,247 @@
+					Selector: &metav1.LabelSelector{
+						MatchLabels: map[string]string{
+							"app": "nginx",
+						},
+					},
+					Template: v1.PodTemplateSpec{
+						ObjectMeta: metav1.ObjectMeta{
+							Labels: map[string]string{
+								"app": "nginx",
+							},
+						},
+						Spec: v1.PodSpec{
+							Containers: []v1.Container{
+								{
+									Name:  "nginx",
+									Image: "nginx:latest",
+									Ports: []v1.ContainerPort{
+										{
+
+											ContainerPort: 80,
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			}
+
+			_, err := client.AppsV1().Deployments("default").Create(deployment)
+			if err != nil {
+				panic(err)
+			}
+			fmt.Println("Deployment created successfully...")
+
+			// creating chaos-experiment for pod-delete
+
+			By("Creating ChaosExperiments")
+			ChaosExperiment := &chaosEngineV1alpha1.ChaosExperiment{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "pod-delete",
+					Namespace: "default",
+					Labels: map[string]string{
+						"litmuschaos.io/name": "kubernetes",
+					},
+				},
+				Spec: chaosEngineV1alpha1.ChaosExperimentSpec{
+					Definition: chaosEngineV1alpha1.ExperimentDef{
+
+						Args:    []string{"-c", "ansible-playbook ./experiments/chaos/pod_delete/test.yml -i /etc/ansible/hosts -vv; exit 0"},
+						Command: []string{"/bin/bash"},
+
+						ENVList: []chaosEngineV1alpha1.ENVPair{
+							{
+								Name:  "ANSIBLE_STDOUT_CALLBACK",
+								Value: "default",
+							},
+							{
+								Name:  "TOTAL_CHAOS_DURATION",
+								Value: "15",
+							},
+							{
+								Name:  "CHAOS_INTERVAL",
+								Value: "5",
+							},
+							{
+								Name:  "LIB",
+								Value: "",
+							},
+						},
+						Image: "",
+						Labels: map[string]string{
+							"name": "pod-delete",
+						},
+						//Litmusbook: "/experiments/chaos/kubernetes/pod_delete/run_litmus_test.yml",
+					},
+				},
+			}
+
+			_, err = clientSet.ChaosExperiments("default").Create(ChaosExperiment)
+			if err != nil {
+				fmt.Println(err)
+			}
+
+			fmt.Println("ChaosExperiment created successfully...")
+
+			//Creating chaosEngine
+
+			By("Creating ChaosEngine")
+			chaosEngine := &chaosEngineV1alpha1.ChaosEngine{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "engine-nginx",
+					Namespace: "default",
+				},
+				Spec: chaosEngineV1alpha1.ChaosEngineSpec{
+					Appinfo: chaosEngineV1alpha1.ApplicationParams{
+						Appns:    "default",
+						Applabel: "app=nginx",
+					},
+
+					Experiments: []chaosEngineV1alpha1.ExperimentList{
+						{
+							Name: "pod-delete",
+						},
+					},
+				},
+			}
+
+			_, err = clientSet.ChaosEngines("default").Create(chaosEngine)
+			if err != nil {
+				fmt.Println(err)
+			}
+
+			fmt.Println("Chaosengine created successfully...")
+			time.Sleep(100 * time.Second)
+
+			_, err = client.CoreV1().Pods("default").Get("engine-nginx-runner", metav1.GetOptions{})
+			Expect(err).To(BeNil())
```

Comment:
Can you add the check for pod state also

Reply:
okay